File: rock63/src/main/java/com/uksusoff/rock63/ui/EventDetailActivity.java
package com.uksusoff.rock63.ui;

import android.text.method.LinkMovementMethod;
import android.view.View;
import android.widget.Button;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.core.app.ShareCompat;

import com.koushikdutta.urlimageviewhelper.UrlImageViewHelper;
import com.uksusoff.rock63.R;
import com.uksusoff.rock63.data.entities.EventItem;
import com.uksusoff.rock63.utils.StringUtils;

import org.androidannotations.annotations.EActivity;
import org.androidannotations.annotations.Extra;
import org.androidannotations.annotations.OptionsItem;
import org.androidannotations.annotations.OptionsMenu;
import org.androidannotations.annotations.ViewById;

import java.sql.SQLException;

@EActivity(R.layout.a_event_detail)
@OptionsMenu(R.menu.menu_detail)
public class EventDetailActivity extends AbstractMenuActivity {

    public static final String EXTRA_ITEM_ID = "eventItem";

    @Extra(EXTRA_ITEM_ID)
    int eventId;

    @ViewById(R.id.event_detail_title)
    TextView title;

    @ViewById(R.id.event_detail_date_and_time)
    TextView dateRange;

    @ViewById(R.id.event_detail_infdetailbtn)
    Button detailButton;

    @ViewById(R.id.event_detail_description)
    TextView description;

    @ViewById(R.id.event_detail_venue_title)
    TextView venueTitle;

    @ViewById(R.id.event_detail_venue_address)
    TextView venueAddress;

    @ViewById(R.id.event_detail_venue_phone)
    TextView venuePhone;

    @ViewById(R.id.event_detail_venue_link)
    TextView venueLink;

    @ViewById(R.id.event_detail_venue_vk)
    TextView venueVk;

    private EventItem event;

    @Override
    protected void init() {
        super.init();

        try {
            event = getHelper().getEventItemsDao().queryForId(eventId);
        } catch (SQLException e) {
            throw new RuntimeException(e);
        }

        title.setText(event.title);
        dateRange.setText(event.getDateRange());

        if (event.getVenueItem() != null) {
            venueTitle.setText(event.getVenueItem().title);

            venueAddress.setMovementMethod(LinkMovementMethod.getInstance());
            venueAddress.setText(event.getVenueItem().getMapAddress());

            if (event.getVenueItem().phone != null) {
                venuePhone.setVisibility(View.VISIBLE);
                venuePhone.setMovementMethod(LinkMovementMethod.getInstance());
                venuePhone.setText(StringUtils.fromHtml(event.getVenueItem().phone));
            } else {
                venuePhone.setVisibility(View.GONE);
            }

            if (event.getVenueItem().url != null) {
                venueLink.setVisibility(View.VISIBLE);
                venueLink.setMovementMethod(LinkMovementMethod.getInstance());
                venueLink.setText(StringUtils.fromHtml(event.getVenueItem().url));
            } else {
                venueLink.setVisibility(View.GONE);
            }

            if (event.getVenueItem().vk != null) {
                venueVk.setVisibility(View.VISIBLE);
                venueVk.setMovementMethod(LinkMovementMethod.getInstance());
                venueVk.setText(StringUtils.fromHtml(event.getVenueItem().vk));
            } else {
                venueVk.setVisibility(View.GONE);
            }

            detailButton.setOnClickListener(v -> {
                View inf = EventDetailActivity.this.findViewById(R.id.event_detail_venue_description);

                if (inf.getVisibility() == View.VISIBLE) {
                    inf.setVisibility(View.GONE);
                    detailButton.setText(R.string.event_show_info_button_title);
                } else {
                    inf.setVisibility(View.VISIBLE);
                    detailButton.setText(R.string.event_hide_info_button_title);
                }
            });

            detailButton.setText(R.string.event_show_info_button_title);

        } else {
            venueTitle.setVisibility(View.GONE);
            detailButton.setVisibility(View.GONE);
        }

        description.setMovementMethod(LinkMovementMethod.getInstance());
        description.setText(StringUtils.fromHtml(event.getDescriptionText()));

        if (event.thumbnailMiddle != null) {
            UrlImageViewHelper.setUrlDrawable(
                    (ImageView) findViewById(R.id.event_detail_image),
                    event.thumbnailMiddle,
                    R.drawable.news_medium_placeholder
            );
        } else {
            ((ImageView) findViewById(R.id.event_detail_image)).setVisibility(View.GONE);
        }
    }

    @OptionsItem(R.id.menu_share)
    void menuShare() {
        startActivity(
                new ShareCompat
                        .IntentBuilder(this)
                        .setType("text/plain")
                        .setSubject(event.getShareSubject())
                        .setText(event.getShareText())
                        .getIntent()
        );
    }
}

File: rock63/src/main/java/com/uksusoff/rock63/ui/EventDetailActivity.kt
package com.uksusoff.rock63.ui

import android.annotation.SuppressLint
import android.os.Build
import androidx.core.app.ShareCompat
import android.text.Html
import android.text.method.LinkMovementMethod
import android.view.View
import android.widget.Button
import android.widget.ImageView
import android.widget.TextView
import androidx.core.text.HtmlCompat
import com.koushikdutta.ion.Ion
import com.uksusoff.rock63.R
import com.uksusoff.rock63.data.entities.Event
import org.androidannotations.annotations.*
import java.sql.SQLException
import java.text.SimpleDateFormat
import java.util.*

@SuppressLint("Registered")
@EActivity(R.layout.a_event_detail)
@OptionsMenu(R.menu.menu_detail)
open class EventDetailActivity : BaseMenuActivity() {

    @Extra(EXTRA_ITEM_ID)
    @JvmField
    protected final var eventId:Int = 0

    @ViewById(R.id.event_detail_placephone)
    protected lateinit var placePhone: TextView
    @ViewById(R.id.event_detail_placelink)
    protected lateinit var placeLink: TextView
    @ViewById(R.id.event_detail_placevklink)
    protected lateinit var placeVkLink: TextView
    @ViewById(R.id.event_detail_placename)
    protected lateinit var placeName: TextView
    @ViewById(R.id.event_detail_placeaddr)
    protected lateinit var placeAddress: TextView
    @ViewById(R.id.event_detail_placeinf)
    protected lateinit var placeInfoContainer: View
    @ViewById(R.id.event_detail_infdetailbtn)
    protected lateinit var placeInfoButton: Button

    @ViewById(R.id.event_detail_title)
    protected lateinit var titleView:TextView
    @ViewById(R.id.event_detail_datentime)
    protected lateinit var dateTimeView:TextView
    @ViewById(R.id.event_detail_description)
    protected lateinit var description:TextView
    @ViewById(R.id.event_detail_image)
    protected lateinit var imageView:ImageView

    private lateinit var event: Event

    override fun init() {
        super.init()
        event = database.events.queryForId(eventId)
        titleView.text = event.title
        val fDate = SimpleDateFormat("dd.MM.yyyy", Locale.getDefault())
        val fTime = SimpleDateFormat("HH:mm", Locale.getDefault())

        event.start?.let { start ->
            event.end?.let { end ->
                dateTimeView.text = String.format(
                        "%s %s - %s %s",
                        fDate.format(start),
                        fTime.format(start),
                        fDate.format(end),
                        fTime.format(end)
                )
            } ?: run {
                dateTimeView.text = String.format(
                        "%s %s",
                        fDate.format(start),
                        fTime.format(start)
                )
            }
        }

        event.place?.let { place ->
            placeName.text = place.name
            placeAddress.text = place.address
            place.phone?.takeIf { it != "" }?.let {phone ->
                placePhone.visibility = View.VISIBLE
                placePhone.movementMethod = LinkMovementMethod.getInstance()
                placePhone.text = HtmlCompat.fromHtml(phone, HtmlCompat.FROM_HTML_MODE_LEGACY)
            } ?: run {
                placePhone.visibility = View.GONE
            }

            place.url?.takeIf { it != "" }?.let { url ->
                placeLink.visibility = View.VISIBLE
                placeLink.movementMethod = LinkMovementMethod.getInstance()
                placeLink.text = HtmlCompat.fromHtml(url, HtmlCompat.FROM_HTML_MODE_LEGACY)
            } ?: run {
                placeLink.visibility = View.GONE
            }

            place.vkUrl?.takeIf { it != "" }?.let { vkUrl ->
                placeVkLink.visibility = View.VISIBLE
                placeVkLink.movementMethod = LinkMovementMethod.getInstance()
                placeVkLink.text = HtmlCompat.fromHtml(vkUrl, HtmlCompat.FROM_HTML_MODE_LEGACY)
            } ?: run {
                placeVkLink.visibility = View.GONE
            }

            placeInfoButton.setText(R.string.event_show_info_button_title)
        } ?: run {
            placeName.visibility = View.GONE
            placeInfoButton.visibility = View.GONE
        }

        description.movementMethod = LinkMovementMethod.getInstance()
        description.text = HtmlCompat.fromHtml(event.body, HtmlCompat.FROM_HTML_MODE_LEGACY)

        event.mediumThumbUrl?.let { url ->
            Ion.with(this)
                    .load(url)
                    .withBitmap()
                    .placeholder(R.drawable.news_medium_placeholder)
                    .intoImageView(imageView)
        } ?: run {
            imageView.visibility = View.GONE
        }
    }

    @Click(R.id.event_detail_infdetailbtn)
    open fun onPlaceInfoButtonClick() {
        if (placeInfoContainer.visibility == View.VISIBLE) {
            placeInfoContainer.visibility = View.GONE
            placeInfoButton.setText(R.string.event_show_info_button_title)
        } else {
            placeInfoContainer.visibility = View.VISIBLE
            placeInfoButton.setText(R.string.event_hide_info_button_title)
        }
    }

    @OptionsItem(R.id.menu_share)
    fun menuShare() {
        shareEvent()
    }

    private fun shareEvent() {
        val subject: String?
        subject = event.place?.let { place ->
            String.format("%s @ %s", event.title, place.name)
        } ?: event.title

        startActivity(ShareCompat.IntentBuilder.from(this)
                .setType("text/plain")
                .setSubject(subject)
                .setText(event.url)
                .intent
        )
    }

    companion object {
        const val EXTRA_ITEM_ID = "eventItem"
    }
}